pinot-common/src/main/proto/worker.proto

```diff
@@ -70,23 +70,17 @@ message QueryResponse {
 
 message StagePlan {
   int32 stageId = 1;
-  string instanceId = 2;
+  string virtualAddress = 2;
   StageNode stageRoot = 3;
-  map<int32, StageMetadata> stageMetadata = 4;
+  repeated StageMetadata stageMetadata = 4;
 }
 
 message StageMetadata {
-  repeated string instances = 1;
-  repeated string dataSources = 2;
-  map<string, SegmentMap> instanceToSegmentMap = 3;
-  string timeColumn = 4;
-  string timeValue = 5;
+  repeated WorkerMetadata workerMetadata = 1;
+  map<string, string> customProperty = 2;
 }
 
-message SegmentMap {
-  map<string, SegmentList> tableTypeToSegmentList = 1;
-}
-
-message SegmentList {
-  repeated string segments = 1;
+message WorkerMetadata {
+  string virtualAddress = 1;
+  map<string, string> customProperty = 2;
```

Comment:
it will be good to split this into 2. metadata that's global.. across all stages and another per stageNode.

Reply:
there's no per stage node metadata, (they will be associated with each stage Node).
the split is per-stage and per-worker